File: src/main/java/vazkii/botania/common/item/equipment/bauble/ItemBalanceCloak.java
// This is here since ItemHolyCloak#getRenderTexture is package-private
package vazkii.botania.common.item.equipment.bauble;

import alfheim.AlfheimCore;
import alfheim.api.lib.LibResourceLocations;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.DamageSource;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.event.entity.living.LivingHurtEvent;
import vazkii.botania.common.Botania;

public class ItemBalanceCloak extends ItemHolyCloak {
	
	public ItemBalanceCloak() {
		super("BalanceCloak");
		setCreativeTab(AlfheimCore.alfheimTab);
	}
	
	@Override
	public boolean effectOnDamage(LivingHurtEvent event, EntityPlayer player, ItemStack stack) {
		if(!event.source.isMagicDamage()) {
			event.ammount /= 2;
			
			if(event.source.getEntity() != null)
				event.source.getEntity().attackEntityFrom(DamageSource.causeIndirectMagicDamage(player, player), event.ammount);
			
			if(event.ammount > player.getHealth())
				event.ammount = player.getHealth() - 1;
			
			player.worldObj.playSoundAtEntity(player, "botania:holyCloak", 1F, 1F);
			
			for(int i = 0; i < 30; i++) {
				double x = player.posX + Math.random() * player.width * 2 - player.width;
				double y = player.posY + Math.random() * player.height;
				double z = player.posZ + Math.random() * player.width * 2 - player.width;
				boolean green = Math.random() > 0.5;
				Botania.proxy.sparkleFX(player.worldObj, x, y, z, 0.3F, green ? 1F : 0.3F, green ? 0.3F : 1F, 0.8F + (float) Math.random() * 0.4F, 3);
			}
			return true;
		}

		return false;
	}
	
	@Override
	public ResourceLocation getRenderTexture() {
		return LibResourceLocations.balanceCloak;
	}
}

File: src/main/java/alfheim/common/item/equipment/bauble/ItemBalanceCloak.kt
package alfheim.common.item.equipment.bauble

import alexsocol.asjlib.*
import alfheim.api.lib.LibResourceLocations
import alfheim.common.core.util.AlfheimTab
import cpw.mods.fml.relauncher.*
import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.*
import net.minecraftforge.event.entity.living.LivingHurtEvent
import vazkii.botania.client.core.helper.IconHelper
import vazkii.botania.common.Botania
import vazkii.botania.common.item.equipment.bauble.ItemHolyCloak

class ItemBalanceCloak: ItemHolyCloak("BalanceCloak") {
	
	init {
		creativeTab = AlfheimTab
	}
	
	@SideOnly(Side.CLIENT)
	override fun registerIcons(reg: IIconRegister) {
		iconHoly = IconHelper.forName(reg, "cloak_holy")
		iconUnholy = IconHelper.forName(reg, "cloak_unholy")
		itemIcon = IconHelper.forName(reg, "cloak_balance")
	}
	
	override fun effectOnDamage(event: LivingHurtEvent, player: EntityPlayer, stack: ItemStack?): Boolean {
		if (!event.source.isMagicDamage) {
			event.ammount /= 2f
			
			if (event.source.entity != null)
				event.source.entity.attackEntityFrom(DamageSource.causeIndirectMagicDamage(player, player), event.ammount)
			
			if (event.ammount > player.health)
				event.ammount = player.health - 1
			
			player.playSoundAtEntity("botania:holyCloak", 1f, 1f)
			
			for (i in 0..29) {
				val x = player.posX + Math.random() * player.width.D * 2.0 - player.width
				val y = player.posY + Math.random() * player.height
				val z = player.posZ + Math.random() * player.width.D * 2.0 - player.width
				val green = Math.random() > 0.5
				Botania.proxy.sparkleFX(player.worldObj, x, y, z, 0.3f, if (green) 1f else 0.3f, if (green) 0.3f else 1f, 0.8f + Math.random().F * 0.4f, 3)
			}
			return true
		}
		
		return false
	}
	
	@SideOnly(Side.CLIENT)
	override fun getRenderTexture() = LibResourceLocations.cloakBalance
	
	companion object {
		lateinit var iconHoly: IIcon
		lateinit var iconUnholy: IIcon
	}
}